WorkflowTesting/Sources/WorkflowActionTester.swift

```diff
@@ -209,10 +227,25 @@ struct TestApplyContext<Wrapped: Workflow>: ApplyContextType {
         case .workflow(let workflow):
             return workflow[keyPath: keyPath]
         case .expectations(var expectedValues):
-            guard let value = expectedValues.removeValue(forKey: keyPath) as? Value else {
```

Comment:
without separating the dictionary lookup from the cast, if you didn't pass in a `Workflow` instance to the `.tester()` API and the `apply()` implementation was reading a value of optional type, this line would always 'pass' and return `nil` when we really want it to fail with the informative error messages below.